File: 1550. Three Consecutive Odds/threeConsecutiveOdds.cpp
#include <iostream>
#include <vector>
using namespace std;


class Solution {
    public:
        bool threeConsecutiveOdds(vector<int>& arr) {
            int cnt = 0;
    
            for(auto num : arr){
                //base case
                if(cnt == 3)return true;
                if(num % 2 == 0){
                    cnt = 0;
                }
                else{
                    cnt++;
                }
            }
    
            return cnt == 3;
        }
    };

File: 1550. Three Consecutive Odds/threeConsecutiveOdds.ts
function threeConsecutiveOddss(arr: number[]): boolean {
    let cnt = 0

    for(let num of arr){
        //base case
        if(cnt === 3)return true;
        cnt = num % 2 === 0 ? 0 : cnt + 1;
    }
    return cnt === 3;
};